packages/amplify-graphql-relational-transformer/src/has-one/has-one-directive-transformer-factory.ts

```diff
@@ -18,6 +21,22 @@ export const getHasOneDirectiveTransformer = (
     case 'POSTGRES':
       return hasOneDirectivePostgresTransformer;
     case 'DYNAMODB':
+      // If references are passed to the directive, we'll use the references relational
+      // modeling approach.
+      if (config.references) {
+        // Passing both references and fields is not supported.
+        if (config.fields) {
+          // TODO: Better error message
+          throw new Error('Something went wrong >> cannot have both references and fields.');
+        }
+        if (config.references.length < 1) {
+          throw new Error(`Invalid @hasMany directive on ${config.field.name.value} - empty references list`);
+        }
```

Comment:
would simplify this nested logic here

Reply:
I'm open to suggestions, do you have one?

We have a few states that we need to check to provide applicable error messages:

```mermaid
flowchart

REFARG{`references` arg passed?}
REFARG --> |yes| FIELDSARG
REFARG --> |no| FIELDS[`fields` behavior]
REFEMPTY{`references` arg empty?}
REFEMPTY --> |yes| E1[Error: `references` cannot be empty]
REFEMPTY --> |no| REF[`references` behavior]
FIELDSARG{`fields` arg passed?}
FIELDSARG --> |yes| E2[Error: cannot pass `references` and `fields`]
FIELDSARG --> |no| REFEMPTY
```

Reply:
This is one option, but I don't see how it simplifies anything.
```ts
if (config.references && config.fields) { // error }
if (config.references) {
  if (config.references.length > 0) { // references }
  // error 
}
// fields
```
another option:
```ts
if (config.references && config.fields) { // error }
if (config.references?.length > 0) { // references }
// fields
```
would reduce the complexity. But it does so at the cost of making this `@hasOne(references: [])` use `fields` behavior, which we really don't want.

Reply:
```
if (config.references && config.fields) { // error }
if (config.references && config.references.length <= 0) { //error }

if (config.references) { // references }
// fields
```
Is what I had in mind so the error conditions are upfront and easy to read, but this is nit picky and not worth a revision.